homeassistant/components/plugwise/sensor.py

```diff
@@ -310,6 +317,8 @@ def __init__(self, api, coordinator, name, dev_id, sensor):
     @property
     def icon(self):
         """Return the icon to use in the frontend."""
+        if CUSTOM_ICONS.get(self._sensor) is not None:
+            self._icon = CUSTOM_ICONS.get(self._sensor)
```

Comment:
Please set the icon attribute in `_async_process_data` instead so we have all updates to that attribute in one place.

Reply:
Can be done but this will lead to two more lines of code as the above two lines will need to be added in the Classes PwThermostatSensor and PwPowerSensor.

Reply:
Added as indicated by @MartinHjelmare  (with the repetition of code) - as @bouwew stated if it should by DRY then I'll revert it.